plugins/source/hubspot/client/spec.go

```diff
@@ -1,7 +1,42 @@
 package client
 
 type Spec struct {
-	MaxRequestsPerSecond *int `yaml:"max_requests_per_second,omitempty" json:"max_requests_per_second,omitempty"`
+	MaxRequestsPerSecond *int          `yaml:"max_requests_per_second,omitempty" json:"max_requests_per_second,omitempty"`
+	Companies            CompaniesSpec `yaml:"companies,omitempty" json:"companies,omitempty"`
```

Comment:
I feel like the following could be done instead:
```go
type TableOptions struct {
	Properties []string `yaml:"properties,omitempty" json:"properties,omitempty"`
}
```

And the spec would just have an extra field:
```go
TableOptions map[string]*TableOptions `yaml:"table_options,omitempty" json:"table_options,omitempty"`
```

This way in the spec you would just have:
```yaml
table_options:
  hubspot_crm_companies:
  - property1
  - property2
  hubspot_crm_deals:
  - deal_property
```

Additionally, it will make this code much more extendable and easy to maintain.

Reply:
@fsschmitt thank you for the PR!
Could you evaluate the `TableOptions` struct I propose in the comment above?

Reply:
> it would allow to extend each entity specs to include

That's one of the reasons why I proposed `TableOptions` struct instead of `map[string][]string` or something similar.

I do understand that HubSpot resources can have specific options for the API calls, but I just wanted to weigh in the possibility to use the easier approach until the specific table options are being implemented and do not fit into the common table options pattern.

I'd also love to get an input from @yevgenypats before going forward in any of those directions.

Reply:
Yeah I agree with @candiduslynx here. I would go with something of the above:

```
type TableOptions struct {
  properties []string
  propertiesWithHistory []string
}
```

as it seems all tables has the same options given all hubspot tables have those methods. If there will be tables that don't have those methods we can always add the validation to exclude those.

I think the `TableOptions` is hubspot specific as you said but it's just mentioned in the internal spec which is fine. We can also name it `api_options` or any other name would work but I think we should use one dynamic map config instead of static per table.

Reply:
Understood, makes total sense as all of them (except pipeline and owners) would follow the same pattern we can make them under the same struct.

Nice catch and thank you for the suggestion! I'll update it right now.